downstairs/src/extent_inner_sqlite.rs

```diff
@@ -342,12 +342,37 @@ impl SqliteMoreInner {
                 (job_id.0, self.extent_number, n_contiguous_blocks as u64)
             });
 
-            nix::sys::uio::pread(
```

Comment:
For some reason, the SQLite backend wasn't checking the returned number of bytes here.